engine/access/access_test.go

```diff
@@ -795,13 +791,11 @@ func (suite *Suite) TestGetSealedTransaction() {
 		suite.request.On("EntityByID", collection.ID(), mock.Anything).Return()
 		// 4. Indexer IndexCollection receives the requested collection and all the execution receipts
 		// Create a lock context for indexing
-		indexLctx := suite.lockManager.NewContext()
-		lockErr := indexLctx.AcquireLock(storage.LockInsertCollection)
-		require.NoError(suite.T(), lockErr)
-		defer indexLctx.Release()
-
-		err = indexer.IndexCollection(indexLctx, collection, collections, suite.log, module.CollectionExecutedMetric(collectionExecutedMetric))
-		require.NoError(suite.T(), err)
+		unittest.WithLock(suite.T(), suite.lockManager, storage.LockInsertCollection, func(indexLctx lockctx.Context) error {
+			err := indexer.IndexCollection(indexLctx, collection, collections, suite.log, module.CollectionExecutedMetric(collectionExecutedMetric))
```

Comment:
delegate error handling (or pass through, if you happen to go with my suggestion) to `WithLock`